src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/avoiddoublebraceinitialization/InputAvoidDoubleBraceInitialization.java

```diff
@@ -0,0 +1,103 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.avoiddoublebraceinitialization;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/*
+ * Config = default
+ */
+public class InputAvoidDoubleBraceInitialization {
+    void m() {
+        ArrayList<?> list = new ArrayList<Object>() { // violation
+            {
+                add(null);
+            }
+        };
+        list = new ArrayList<Object>() { // violation
+            {
+                add(null);
+            }
+
+            {
+                add(null);
+            }
+        };
+        list = new ArrayList<Object>() { // violation
+            // some comment
+            {
+                add(null);
+            }
+        };
+        list = new ArrayList<Object>() { // violation
+            {
+                add(null);
+            } // some comment
+        };
+        list = new ArrayList<Object>() { // violation
+            {
+                add(null);
+            }
+            // some comment
+        };
+        list = new ArrayList<Object>() { // violation
+            /* comment */ {
+                add(null);
+            }
+        };
+        list = new ArrayList<Object>() { // violation
+            {
+                add(null);
+            }
+            /* comment */
+        };
+        list = new ArrayList<Object>() { // violation
+            {
+                add(null);
+            } /* comment */
+        };
+        list = new ArrayList<Object>() {
+            ;
+            ;
+            {}
+            ;
+            ;
+            ;
+        };
+        Object obj = new Serializable() {  // violation
+            {}
```

Comment:
please add one case like this - https://stackoverflow.com/a/1958961/1015848

Reply:
done